### DISCUSSION TITLE
?Ilia?? Launcher // zotero :: get the launcher to list a manually installed package

### DISCUSSION THREAD

**Discussion by Whyglobaleyes** [regolith-linux/regolith-desktop#696] [General]
Evening All,
I really am out of my depth with this. I've always let the automagics handle installation or cludged a work-around with an alias but that isn't working this time. Apologies in advance if this is the wrong place to ask. 

I've installed Zotero by following instructions here: https://www.zotero.org/support/installation ... the package is installed in `/opt/zotero` and I've got a symlink entry at `~/.local/share/applications/zotero.desktop`. 
- When I try to launch Zotero from Ilia (that's replace rofi hasn't it?) then the programme isn't showing up. 
- `which zotero` doesn't find it but ...  
- I can launch it by entering the full filepath in a terminal `/opt/zotero/Zotero_linux-x86_64/zotero`. 
- I made a guess and put a copy of `zotero.desktop` in `~/.local/share/desktop-directories` but that made no difference.

Anyone know what I should do please?
TIA

**Comment by ferdinandyb** [comment]
In `~/.local/share/applications/zotero.desktop` you are symlinking to `/opt/zotero/zotero.desktop`? I have the exact same setup and zotero shows up in ilia, albeit without the icon. And there's actually an executable at `/opt/zotero/zotero` maybe you could try instead of the symlink a copy and changing the path to hardcode that? That Exec line in the desktop file looks pretty convoluted.

**Comment by Whyglobaleyes** [comment]
Hi @ferdinandyb, Thanks for your advice. 

1. I had created my `~/.local/share/applications/zotero.desktop` entry by copying the file from `/opt/zotero/Zotero_linux-x86_64/zotero.desktop` (my installation went into a slightly different directory in `/opt/`). The `~/.local ...desktop` file successfully launches zotero when I click on it in Thunar, but it doesn't appear in Ilia.
2. After your suggestion I deleted the copied f.desktop file and created a symlink `~/.local/share/applications/zotero.desktop` pointing `/opt/zotero/Zotero_linux-x86_64/zotero.desktop` ==> that behaves just like (1).
3. I noticed that the symlinked file was owned by `root:root` and thought that might be hiding it from ilia so I tried `sudo chown -R myUser:myGroup /opt/zotero/` but that made no difference.
4. I agree that the exec line is horrible. So I deleted the symlink and created a new `~/.local/share/applications/zotero.desktop` with the contents below. It behaves exactly the same as (1). 

- My only remaining idea (for now) is that perhaps something in my setup is preventing ilia from seeing items in my `~/.local/share/applications`. 
- I had a look at https://github.com/regolith-linux/ilia#desktop-apps-apps and noticed that it mentions _"the `XDG_DATA_DIRS` environment variable"_
- That env_variable isn't set on my system. So I tried creating it manually by `export XDG_DATA_DIRS=~/.local/share/applications/`. That didn't make any difference

So I'm back to square one and at a loss again.
If you've got any other ideas then I'd be very grateful. If not then I might try to back up my zotero settings then do a fresh re-install on it 

thanks once again and TIA

---
```
[Desktop Entry]
Name=Zotero
Exec=/opt/zotero/Zotero_linux-x86_64/zotero
Icon=/opt/zotero/Zotero_linux-x86_64/chrome/icons/default/default256.png
Type=Application
Terminal=false
Categories=Office;
MimeType=text/plain;x-scheme-handler/zotero;application/x-research-info-systems;text/x-research-info-systems;text/ris;application/x-endnote-refer;application/x-inst-for-Scientific-info;application/mods+xml;application/rdf+xml;application/x-bibtex;text/x-bibtex;application/marc;application/vnd.citationstyles.style+xml
X-GNOME-SingleWindow=true
```

**Comment by ferdinandyb** [comment]
I think that will be the problem, mine looks like this:
```
/usr/share/regolith:/usr/share/gnome:/usr/local/share/:/usr/share/:/home/fbence/.local/share
```
so `.local/share` is referenced, not applications directly.

**Comment by kgilmer** [comment]
> That env_variable isn't set on my system. So I tried creating it manually by export XDG_DATA_DIRS=~/.local/share/applications/. That didn't make any difference

Regarding this, the export won't work unless it's done before the X session is initialized.  There has been some trouble found when trying to modify this variable on behalf of the user in Debian.  The plan is to add an option to ilia to allow it to explicitly load from a set of configured directories rather than reading from `XDG_DATA_DIRS`.

**Reply by kgilmer** [reply]
The logic in ilia has been updated to source the desktop app paths from dconf/gsettings rather than env variables.  This allows users to easily update where ilia searches for desktop apps.  See https://github.com/regolith-linux/ilia/commit/f9eacc0c966853ea046a49703f602393e5dfe076 for the changes that will be available in version `0.12.17`.

**Reply by Whyglobaleyes** [reply]
Hi Ken,

Nice to hear from you because I am so appreciative of regolith. Thanks for your attention and explaining ==> I don't really understand the explanation but - **I've got it working now!** so that's fabulous. Thank you very much.
1. Can I ask when 0.12.17 is expected please? 
2. Also - and it feels unworthy to be asking you but my google-fu failed - could you link me to update instructions please? 

--------------

For the (possible) benefit of anyone following I'll describe what worked for me.  If anyone can explain or improve the workaround I've used then I'd be grateful to learn. 

1. I tried adding `/home/USER/.local/share` as well as `.local/share/applications` to my bashrc (as per ferdinandyb 10d ago) ==> that didn't work either - which I guess is because the bashrc is read when I launch a terminal which I suppose is a long time after Xsession is launched.
2. The solution was to edit the `/etc/X11/Xsession` file (sudo req'd and I backed up the original file first) by adding `export XDG_DATA_DIRS=/usr/share/regolith:/usr/share/gnome:/usr/local/share/:/usr/share/:/home/a/.local/share:/home/a/.local/share/applications`.
3. After rebooting ilia recognises zotero!!
4. I suspect that the proper way to modify the Xsession isn't to directly edit `/etc/X11/Xsession` but I wasn't able to figure out what directory/ file to modify or what syntax to use
